common/scala/src/main/scala/whisk/core/database/CouchDbRestStore.scala

```diff
@@ -357,22 +359,58 @@ class CouchDbRestStore[DocumentAbstraction <: DocumentSerializer](dbProtocol: St
     docStream: Source[ByteString, _],
     oldAttachment: Option[Attached])(implicit transid: TransactionId): Future[(DocInfo, Attached)] = {
 
-    for {
-      (bytes, tailSource) <- inlineAndTail(docStream)
-      uri <- Future.successful(uriOf(bytes, UUID().asString))
-      attached <- {
-        val a = if (isInlined(uri)) {
-          Attached(uri.toString, contentType, Some(bytes.size), Some(digest(bytes)))
-        } else {
-          Attached(uri.toString, contentType)
-        }
-        Future.successful(a)
-      }
-      i1 <- put(update(d, attached))
-      i2 <- if (isInlined(uri)) { Future.successful(i1) } else {
-        attach(i1, uri.path.toString, attached.attachmentType, combinedSource(bytes, tailSource))
-      }
-    } yield (i2, attached)
+    attachmentStore match {
+      case Some(as) =>
+        val asJson = d.toDocumentRecord
```

Comment:
do yo mind factoring the code for some and none into their own methods?
a comment on what happens if no store is defined would be helpful otherwise (code is inlined).

Reply:
Done